featurebyte/worker/task/catalog_cleanup.py

```diff
@@ -0,0 +1,230 @@
+from featurebyte.service.feature_store import FeatureStoreService
+from featurebyte.service.session_manager import SessionManagerService
+from featurebyte.service.task_manager import TaskManager
+from featurebyte.storage import Storage
+from featurebyte.worker.task.base import BaseTask
+from featurebyte.worker.util.task_progress_updater import TaskProgressUpdater
+
+
+def get_catalog_subclasses_in_module(
+    module: Any,
+) -> list[Type[FeatureByteCatalogBaseDocumentModel]]:
+    """
+    Retrieve strict subclasses of FeatureByteCatalogBaseDocumentModel defined in a given module.
+
+    Parameters
+    ----------
+    module : Any
+        module to search for subclasses
+
+    Returns
+    -------
+    list[Type[FeatureByteCatalogBaseDocumentModel]]
+        list of subclasses of FeatureByteCatalogBaseDocumentModel
+    """
+    classes = []
+    for name, obj in inspect.getmembers(module, inspect.isclass):
+        # Ensure the class is defined in this module and is a strict subclass of base_class
+        if (
+            obj.__module__ == module.__name__
+            and issubclass(obj, FeatureByteCatalogBaseDocumentModel)
+            and hasattr(obj, "collection_name")
+        ):
+            try:
+                obj.collection_name()
+                classes.append(obj)
+            except AttributeError:
+                print(name, obj)
+                pass
+    return classes
+
+
+class CatalogCleanupTask(BaseTask[CatalogCleanupTaskPayload]):
+    """
+    Catalog Cleanup Task
+    """
+
+    payload_class = CatalogCleanupTaskPayload
+
+    def __init__(
+        self,
+        task_manager: TaskManager,
+        feature_store_service: FeatureStoreService,
+        all_catalog_service: AllCatalogService,
+        persistent: Persistent,
+        storage: Storage,
+        session_manager_service: SessionManagerService,
+        task_progress_updater: TaskProgressUpdater,
+    ):
+        super().__init__(task_manager=task_manager)
+        self.task_progress_updater = task_progress_updater
+        self.feature_store_service = feature_store_service
+        self.all_catalog_service = all_catalog_service
+        self.persistent = persistent
+        self.storage = storage
+        self.session_manager_service = session_manager_service
+
+    @property
+    def model_packages(self) -> list[Any]:
+        """
+        Model packages to look for catalog specific models
+
+        Returns
+        -------
+        list[Any]
+            list of model packages
+        """
+        return [featurebyte.models]
+
+    @property
+    def remap_model_classes(self) -> dict[str, Type[FeatureByteCatalogBaseDocumentModel]]:
+        """
+        Remap model classes to cleanup
+
+        Returns
+        -------
+        dict[str, Type[FeatureByteCatalogBaseDocumentModel]]
+            remap model classes
+        """
+        return {"table": ProxyTableModel}
+
+    @property
+    def catalog_specific_model_class_pairs(
```

Comment:
Would it make sense to make this a cached property since the list is being derived repeatedly for every processed catalog if I understand correctly?

Reply:
Good catch! Will change this to cached property.